code/game/objects/items/RCD.dm

```diff
@@ -1157,56 +1182,152 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
-		var/obj/machinery/plumbing/P = A
-		if(P.anchored)
-			to_chat(user, span_warning("The [P.name] needs to be unanchored!"))
+	if(istype(target, /obj/machinery/plumbing))
+		var/obj/machinery/machine_target = target
+		if(machine_target.anchored)
+			to_chat(user, span_warning("The [target.name] needs to be unanchored!"))
 			return
-		if(do_after(user, 20, target = P))
-			P.deconstruct() //Let's not substract matter
+		if(do_after(user, 20, target = target))
+			machine_target.deconstruct() //Let's not substract matter
 			playsound(get_turf(src), 'sound/machines/click.ogg', 50, TRUE) //this is just such a great sound effect
 	else
-		create_machine(A, user)
+		create_machine(target, user)
 
 /obj/item/construction/plumbing/AltClick(mob/user)
-	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE))
+	// give a menu to pick layers or colors
+	var/list/options_menu = list(
+		"Current Layer" = image('icons/hud/radial.dmi', icon_state = "plumbing_layer[GLOB.plumbing_layers[current_layer]]"),
+		"Current Color" = image('icons/hud/radial.dmi', icon_state = current_color),
+	)
+
+	playsound(src.loc, 'sound/effects/pop.ogg', 50, FALSE)
+	var/choice = show_radial_menu(user, src, options_menu, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!check_menu(user))
 		return
 
-	//this is just cycling options through a list
-	var/current_loc = layers.Find(current_layer) + 1
+	switch(choice)
+		if("Current Layer")
+			choose_layer_menu(user)
+		if("Current Color")
+			choose_color_menu(user)
 
-	if(current_loc > layers.len)
-		current_loc = 1
+/**
+ * Choose the current layer via radial menu.
+ *
+ * Arguments:
+ * * user - current user.
+ */
+/obj/item/construction/plumbing/proc/choose_layer_menu(mob/user)
+	if(!GLOB.plumbing_layer_menu_options.len)
+		for(var/layer_name in GLOB.plumbing_layers)
+			GLOB.plumbing_layer_menu_options += list((layer_name) = image('icons/hud/radial.dmi', icon_state = "plumbing_layer[GLOB.plumbing_layers[layer_name]]"))
 
-	//We want the key (the define), not the index (the string)
-	current_layer = layers[current_loc]
-	to_chat(user, span_notice("You switch [src] to [current_layer]."))
+	playsound(src.loc, 'sound/effects/pop.ogg', 50, FALSE)
+	var/new_layer = show_radial_menu(user, src, GLOB.plumbing_layer_menu_options, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!new_layer || !check_menu(user))
+		return
+
+	current_layer = new_layer
+	to_chat(user, span_notice("You set the layer to [new_layer]."))
+
+/**
+ * Choose the current color via radial menu.
+ *
+ * Arguments:
+ * * user - current user.
+ */
+/obj/item/construction/plumbing/proc/choose_color_menu(mob/user)
+	if(!GLOB.plumbing_color_menu_options.len)
+		for(var/color_name in GLOB.pipe_paint_colors)
+			GLOB.plumbing_color_menu_options += list((color_name) = image('icons/hud/radial.dmi', icon_state = color_name))
+
+	playsound(src.loc, 'sound/effects/pop.ogg', 50, FALSE)
```

Comment:
.

Reply:
Replaced *all* instanced of `src.loc` in rcd.dm to loc since these playsound lines are just copy-pastes in the first place.